examples/binance-futures-usdm.ts

```diff
@@ -0,0 +1,357 @@
+    
+    // Clear existing positions and orders from state
+    state.getAllPositions().forEach(pos => {
+      state.deleteActivePosition(pos.symbol, pos.positionSide);
+    });
+    state.clearAllOrders();
+    
+    // Fetch account information to get balance
+    const accountInfo = await restClient.getAccountInformationV3();
+    
+    if (accountInfo && accountInfo.totalWalletBalance) {
+      state.setWalletBalance(Number(accountInfo.totalWalletBalance));
+      console.log(new Date(), `Updated wallet balance: ${accountInfo.totalWalletBalance} USDT`);
+    }
+    
+    // Fetch all positions
+    const positions = await restClient.getPositionsV3();
+    
+    // Filter active positions (non-zero position amount)
+    const activePositions = positions.filter(pos => Number(pos.positionAmt) !== 0);
+    
+    // Update state with active positions
+    activePositions.forEach(pos => {
+      const enginePos = mapBinanceRestPositionToEnginePosition(pos);
+      state.setActivePosition(pos.symbol, enginePos.positionSide, enginePos);
+      console.log(new Date(), `Position updated for ${pos.symbol} ${enginePos.positionSide}: ${pos.positionAmt} @ ${pos.entryPrice} (UPNL: ${pos.unRealizedProfit})`);
+    });
+
+    // Fetch all open orders
+    const openOrders = await restClient.getAllOpenOrders();
+    
+    // Update state with open orders
+    openOrders.forEach(order => {
+      const engineOrder = mapBinanceRestOrderToEngineOrder(order);
+      
+      state.upsertActiveOrder(engineOrder);
+      console.log(new Date(), `Order synced: ${order.symbol} ${order.side} ${order.type} ${order.status}`);
+    });
+    
+    // Summary of active positions and orders
+    const activePositionCount = state.getTotalActivePositions();
+    const activeOrders = state.getActiveOrders();
+    console.log(new Date(), `Sync complete. Positions: ${activePositionCount.total}, Active Orders: ${activeOrders.length}`);
+    
+    return true;
+  } catch (error) {
+    console.error(new Date(), 'Error syncing from REST API:', error);
+    return false;
+  }
+}
+
+/**
+ * Main function to start the application
+ */
+async function main() {
+  console.log(new Date(), 'Starting Binance USDM Futures position tracking...');
+  
+  // Initial sync from REST API
+  await syncPositionsFromREST();
+  
+  // Setup WebSocket event handlers
+  wsClient.on('open', (data) => {
+    console.log(new Date(), `WebSocket connection opened: ${data.wsKey}`);
+  });
+  
+  wsClient.on('formattedMessage', (data) => {
+    if (isWsFormattedFuturesUserDataEvent(data)) {
+      // Handle different types of user data events
+      if (isWsFormattedFuturesUserDataAccountUpdate(data)) {
+        handleAccountUpdate(data);
+      } else if (isWsFormattedFuturesUserDataTradeUpdateEvent(data)) {
+        handleOrderUpdate(data);
+      } else {
+        console.log(new Date(), 'Other user data event:', data.eventType);
+      }
```

Comment:
```suggestion
      if (isWsFormattedFuturesUserDataAccountUpdate(data)) {
        handleAccountUpdate(data);
        return;
      } 
      
      if (isWsFormattedFuturesUserDataTradeUpdateEvent(data)) {
        handleOrderUpdate(data);
        return;
      } 
      
      console.log(new Date(), 'Other user data event:', data.eventType);
```

it's pretty much the same as what you did, but I've learnt to love early returns when there's nothing else to do, and to avoid long if else if else chains